xbmc/addons/FontResource.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+*      Copyright (C) 2005-2017 Team XBMC
+*      http://kodi.tv
+*
+*  This Program is free software; you can redistribute it and/or modify
+*  it under the terms of the GNU General Public License as published by
+*  the Free Software Foundation; either version 2, or (at your option)
+*  any later version.
+*
+*  This Program is distributed in the hope that it will be useful,
+*  but WITHOUT ANY WARRANTY; without even the implied warranty of
+*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+*  GNU General Public License for more details.
+*
+*  You should have received a copy of the GNU General Public License
+*  along with Kodi; see the file COPYING.  If not, see
+*  <http://www.gnu.org/licenses/>.
+*
+*/
+#include "FontResource.h"
+
+#include "AddonManager.h"
+#include "dialogs/GUIDialogKaiToast.h"
+#include "filesystem/File.h"
+#include "filesystem/SpecialProtocol.h"
+#include "guilib/GUIWindowManager.h"
+#include "messaging/ApplicationMessenger.h"
+#include "messaging/helpers/DialogHelper.h"
+#include "ServiceBroker.h"
+#include "settings/Settings.h"
+
+using namespace XFILE;
+using namespace KODI::MESSAGING;
+
+using KODI::MESSAGING::HELPERS::DialogResponse;
+
+namespace ADDON
+{
+
+std::unique_ptr<CFontResource> CFontResource::FromExtension(CAddonInfo addonInfo, const cp_extension_t* ext)
+{
+  return std::unique_ptr<CFontResource>(new CFontResource(std::move(addonInfo)));
+}
+
+bool CFontResource::IsInUse() const
+{
+  std::string skin = CServiceBroker::GetSettings().GetString(CSettings::SETTING_LOOKANDFEEL_SKIN);
+  ADDONDEPS deps = CServiceBroker::GetAddonMgr().GetDepsRecursive(skin);
+  for (const auto& it : deps)
+    if (it.first == ID())
+      return true;
+
+  return false;
+}
```

Comment:
Also, this method is redundant. Dependency check is already done elsewhere for all addons

Reply:
i do not understand how this can be redundant, i need to check if the font is in use on update, and reload skin if it is. there are glyph caches and such in memory that has to be redone, and the only way to do that right now is to reload the skin.

Reply:
right, i see where the confusion comes from now. i have been a little bit too eager when i c&p'd the code from the skin, and that dialog shouldn't be there at all. i thought it was something different, not the want to switch dialog.

Reply:
Nah, my confusion was from IsInUse being implemented.. But I get it now, so can you please move this into OnPostInstall instead then? Because `IsInUse` is used by external stuff which means the check will done multiple times. (and IsInUse should really be removed in general)

Reply:
done.